python/grass/temporal/metadata.py

```diff
@@ -1261,15 +1249,9 @@ def print_info(self):
         print(" | Maximum value min:.......... " + str(self.get_max_min()))
         print(" | Maximum value max:.......... " + str(self.get_max_max()))
         print(" | Aggregation type:........... " + str(self.get_aggregation_type()))
-        print(" | Number of registered bands:. " + str(self.get_number_of_bands()))
-
-        STDSMetadataBase.print_info(self)
 
     def print_shell_info(self):
         """Print information about this class in shell style"""
-        print("aggregation_type=" + str(self.get_aggregation_type()))
-        print("number_of_bands=" + str(self.get_number_of_bands()))
-        STDSMetadataBase.print_shell_info(self)
```

Comment:
The way `STDSMetadataBase.print_shell_info(self)` is now called is highly unusual. Please document the reasoning in the code.

Reply:
The way how `STDSMetadataBase.print_shell_info(self)` is called haven't changed. I just moved call from `RasterMetadataBase` to `RasterMetadata` and `Raster3DMetadata` in order to keep printing variables in consistent order.

@wenzeslaus Please explain what you mean by "unusual".

Reply:
As far as I can tell, before, a function from a parent class was called (by a direct child). Now, classes call functions of the their grandparents. I think that's unusual (and [possibly also considered a bad practice](https://stackoverflow.com/a/43943861/574907)).

Reply:
Well, not everything what is mentioned in stackoverflow I would write in the stone. @wenzeslaus Anyway, I took your valid point into consideration.

* `STDSMetadataBase.print_info()` renamed to `print_info_tail()`
* the code changed in order to use `super()` syntax

See https://github.com/OSGeo/grass/pull/1447/commits/57fe78fb867abf865fba2d31b8d029a720788f08

Reply:
> not everything what is mentioned in stackoverflow I would write in the stone

So I though about it again and you can show it just on the standard inheritance example like Animal-Mammal-Dog or Fruit-Apple-Gala. Instances of Dog are Mammals and because Mammals are Animals dogs are Animals (but Dogs are not Animals apart from being Mammals - it is not multiple inheritance). What is relevant here is that, Mammals must be fully Animals, so if there is a `print_attributes()` function in Animal for printing attributes, `print_attributes()` of Mammal is expected to give you attributes from Animal (at least) and (possibly) Mammal (if there are any extra). It is expected because Mammal is an Animal. In this context, Dog should not combine attributes from Animal and attributes from Mammal in order to get some "full Mammal list", the Mammal's attributes contain the Animal attributes because Mammal is an Animal.

> STDSMetadataBase.print_info() renamed to print_info_tail()

I'm afraid that the Animal-Mammal-Dog inheritance example reveals that although this avoids the somewhat evident issue of calling parent and grandparent, it still retains the less apparent issue of not keeping inheritance semantics of children classes being also valid parent classes. Namely, both `STDSRasterMetadataBase` (Mammal) and `STRDSMetadata` (Dog) have `print_info()`, but `STDSRasterMetadataBase.print_info()` does not print the "tail" part while the `STRDSMetadata.print_info()` does. That's like the Mammal not being an Animal and only when we get to Dog, we can see the Animal (`STDSMetadataBase`) there. 

In this class hierarchy, starting with what you did so far, I think the right solution would be having `print_tail` and also `print_body` (and `print_head` or whatever is needed). Each class would define those and there would be default implementation for `print_info` which would be defined as printing all and overwritten when needed.

Perhaps a simpler way to avoid this is to say there is difference between the `*Base` ("abstract") classes and the "final"/"concrete" classes. The base classes define only the building block functions and it is responsibility of the concrete classes to define `print_info`. (In C++, `print_info` would be pure virtual.) So, maybe you could just rename `print_info` in `STDSRasterMetadataBase`.

> the code changed in order to use super() syntax

That's super!